### PR TITLE
Network proxy fn

### PR DESCRIPTION
**Pull request by mfornet**
Allow us to build proxy handlers from functions/closures. Solves #799

### CODE REVIEW

File: node/network/src/proxy/predicate.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+use futures::Stream;
+
+use crate::protocol::SimplePackedMessage;
+use crate::proxy::ProxyHandler;
+
+pub struct FnProxyHandler {
+    predicate: fn(SimplePackedMessage) -> Option<SimplePackedMessage>
```
Comment (nearmax) [inline node/network/src/proxy/predicate.rs]: Trait `Fn` is not to be confused with 'fn': https://stackoverflow.com/questions/54181816/function-pointer-vs-fn-trait-object

Also prefer `FnMut` when possible.
```suggestion
    predicate: FnMut(SimplePackedMessage) -> Option<SimplePackedMessage>
```

File: tests/tests_custom_handler.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    thread::sleep(Duration::from_millis(1000));
+
+    // Execute N trials. In each trial we submit a transaction to a random node i, that sends
+    // 1 token to a random node j. We send transaction to node Then we wait for the balance change to propagate by checking
+    // the balance of j on node k.
+    let mut expected_balances = vec![init_balance; num_nodes];
+    let trial_duration = 10000;
+    for trial in 0..num_trials {
+        println!("TRIAL #{}", trial);
```
Comment (nearmax) [inline tests/tests_custom_handler.rs]:
```suggestion
```

Comment (mfornet) [reply tests/tests_custom_handler.rs]: Fixed

File: node/network/src/proxy/predicate.rs (hunk 1 above)
Comment (mfornet) [reply node/network/src/proxy/predicate.rs]: Moved to #803

**Pull request merged**